cmd/dockerd/daemon.go

```diff
@@ -232,6 +222,11 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 
 	cli.d = d
```

Comment:
Probably this line _and_ the `cli.startMetricsServer` should be moved up, otherwise the metrics server may no longer be giving metrics about restarting the swarm containers